sklearn/neighbors/nearest_centroid.py

```diff
@@ -219,19 +226,33 @@ def _partial_fit(self, X, y, classes=None, _refit=False):
         self.centroids_ = self.true_centroids_[self.nk_ != 0]
 
         if self.shrink_threshold:
```

Comment:
I recommend moving the two "deepcopy" calls on lines 192, 193 into the "if self.shrink_threshold" conditional on line 228. This way you only compute them when needed.